### PR TITLE
Handle iOS status bar changes properly

### PR DESCRIPTION
**Pull request by eeue56**
This fixes an issue where the "in call" statusbar would not correctly resize the window. It only resolves the issue if the user uses a `ClientPanel` - normal panels or raw elements do not scale accordingly, as they lack the TopFrame handlers.

## Test

There are two cases for testing, both only apply to iOS apps that have `ClientPanel`

- Start an application with a `ClientPanel` without being in-call
- toggle in-call on and off

- Start an application with `ClientPanel` with being in-call
- toggle in-call on and off

This PR contains:
- [x] Changelog
- [x] Documentation
- [x] Tests

### CODE REVIEW

File: Source/Fuse.Controls.Panels/Backgrounds/TopFrameBackground.uno [hunk 1]
```diff
@@ -49,7 +49,13 @@ namespace Fuse.Controls
 
 		protected override float2 GetContentSize(LayoutParams lp)
 		{
-			if defined(iOS || Android)
+			if defined(iOS) {
+				// on iOS, we always treat the bar as though it's 40 points
+				var height = 40 / AppBase.Current.PixelsPerPoint;
```
Comment (mortoray) [inline Source/Fuse.Controls.Panels/Backgrounds/TopFrameBackground.uno]: Why? This seems odd to have a hard-coded value here.

File: Source/Fuse.Platform/iOS/SystemUI.uno [hunk 2]
```diff
@@ -211,6 +212,23 @@ namespace Fuse.Platform
 
 			var args = new SystemUIWillResizeEventArgs(SystemUIID.TopFrame, reason, endFrame, startFrame, animationDuration, 1);
 
+			Rect newFrameRect;
+			// when the endFrame size is > 40, we are streching to fill the space
+			if (endFrame.Size.Y > 40)
```
Comment (mortoray) [inline Source/Fuse.Platform/iOS/SystemUI.uno]: The magic number 40 again. Use a `const` with a name, and explanation, or why we have this value.

File: Tests/ManualTests/ManualTestingApp/Singles/StatusBar.ux [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<Page Title="Status bar" ux:Class="StatusBar">
+	<InfoStack ux:Key="Info">
+		<p>This page is intended to test the status bar handling when a phone call is recieved with the app open.</p>
+	</InfoStack>
+	<StackPanel Margin="10">
```
Comment (mortoray) [inline Tests/ManualTests/ManualTestingApp/Singles/StatusBar.ux]: I think it'd be good to have something aligned to all the corners of the page. I guess the top frame already has the app header/buttons, but maybe put too circles in the bottom left/right corners to ensure they aren't getting clipped and/or distorted.

File: Source/Fuse.Controls.Panels/Backgrounds/TopFrameBackground.uno (hunk 1 above)
Comment (eeue56) [inline Source/Fuse.Controls.Panels/Backgrounds/TopFrameBackground.uno]: This is also the issue that got flagged last time - the tricky issue is that it's device specific. We can calculate it by getting the height of the top frame, but there is no consistent measurement we can get at runtime that way which will mean this functions as expected.

Comment (mortoray) [inline Source/Fuse.Controls.Panels/Backgrounds/TopFrameBackground.uno]: It just seems very odd that a PR which is supposed to adjust to a dynamic change in the status bar is using a static size for it?!  This also seems not to be future proof, as I suspect Apple isn't guaranteeing a permanent 40pixel height to the status bar.

Comment (mortoray) [inline Source/Fuse.Controls.Panels/Backgrounds/TopFrameBackground.uno]: Wouldn't fixing it at 40 mean we always lose that top bit of the screen, even when it's only 20 in size?

Comment (eeue56) [inline Source/Fuse.Controls.Panels/Backgrounds/TopFrameBackground.uno]: Yep - this PR is very much work in progress. I was exploring alternative ways to calculate the right compensation but hadn't quite discovered it yet

**Comment by mortoray** [comment]
Closing as abandoned.  Other status changes have come since then, so I can't isolate the issue this PR may still address.

**Pull request closed without merging**